crates/services/gas_price_service/src/fuel_gas_price_updater.rs

```diff
@@ -216,16 +242,77 @@ where
             inner,
             l2_block_source,
             metadata_storage,
+            da_source,
         };
         Ok(updater)
     }
+
+    fn validate_block_gas_capacity(
+        &self,
+        block_gas_capacity: u64,
+    ) -> anyhow::Result<NonZeroU64> {
+        NonZeroU64::new(block_gas_capacity)
+            .ok_or_else(|| anyhow!("Block gas capacity must be non-zero"))
+    }
+
+    async fn set_metadata(&mut self) -> anyhow::Result<()> {
+        self.metadata_storage
+            .set_metadata(self.inner.clone().into())
+            .map_err(|err| anyhow!(err))
+    }
+
+    async fn handle_normal_block(
+        &mut self,
+        height: u32,
+        gas_used: u64,
+        block_gas_capacity: u64,
+    ) -> anyhow::Result<()> {
+        let capacity = self.validate_block_gas_capacity(block_gas_capacity)?;
+
+        match &mut self.inner {
+            AlgorithmUpdater::V0(updater) => {
+                updater.update_l2_block_data(height, gas_used, capacity)?;
+            }
+            AlgorithmUpdater::V1(updater) => {
```

Comment:
I think we should just throw an error on this branch for now. Just to be abundantly clear that it's WIP.

Reply:
addressed in 9d68aa1f2